### ISSUE TITLE
Compiler flag -nomixed_str_len_arg deprecated in Intel 2021

### ISSUE THREAD

**Issue by emkemp** [NASA-LIS/LISF#911]
### Bug Description

Currently the LISF build system sets the flag -nomixed_str_len_arg when compiling with Intel Fortran.  On Narwhal (using Intel oneapi_2021.3.0.3219), this generates a warning about deprecation:

ifort: command line warning #10434: option '-nomixed_str_len_arg' use with underscore is deprecated; use '-nomixed-str-len-arg' instead

This is not a problem *yet* but will be soon.  An adjustment to the build system is warranted.

### Steps to Replicate

Easily replicated on Narwhal.  Load Jim Geiger's private module "lisf_74_prgenv_intel_8.1.0".  Then compile LDT off of the support/lisf-557ww-7.4 branch. (Can also use the master branch.)  Frequent warning messages will appear in the standard output.

**Comment by jvgeiger** [comment]
Yes, I agree.   I must first determine which versions of the Intel compilers our partners are using.  I do not want to accidentally break compilation for "older" Intel compilers.

**Comment by emkemp** [comment]
This is still a TODO.

**Comment by emkemp** [comment]
This is still a TODO.
